applications/crossbar/src/modules/cb_rates.erl

```diff
@@ -133,6 +150,23 @@ put(Context) ->
 delete(Context, _RateId) ->
     crossbar_doc:delete(Context).
 
+-spec validate_number(ne_binary(), cb_context:context()) -> cb_context:context().
+validate_number(Phonenumber, Context) ->
+    case length(re:replace(Phonenumber, "[^0-9]", "", [global, {return, list}])) >= 7 of 
+        'true' -> 
+            rate_for_number(re:replace(Phonenumber, "[^0-9]", "", [global, {return, list}]), Context);
```

Comment:
Once you know it is reconcilable, why not use `wnm_util:to_e164/1` to process it into a known format?